### ISSUE TITLE
MenuItems with where cause for a location that doesn't have a menu assigned still returning results

### ISSUE THREAD

**Issue by ccorda** [wp-graphql/wp-graphql#2722]
### Description

I have multiple menu locations, and after a menu is created and assigned to one location, all locations are returning results even though they don't have menus assigned:

<img width="712" alt="Screenshot 2023-02-12 at 5 33 31 PM" src="https://user-images.githubusercontent.com/45271/218351702-63dcf681-07b4-46a0-87e5-979e9cac6ab8.png">

<img width="1043" alt="Screenshot 2023-02-12 at 5 38 01 PM" src="https://user-images.githubusercontent.com/45271/218351864-095e0137-b1b2-49e0-862c-55a7346442f0.png">

This has come up before in closed issues, but I'm still seeing it as an issue with the latest version of wp-graphql.

https://github.com/wp-graphql/wp-graphql/issues/1822#issuecomment-1307085428
https://github.com/wp-graphql/wp-graphql/issues/2084

### Steps to reproduce

1. Have multiple menu locations.
2. Create a menu and assign it to one location, but leave the others without a menu assigned.
3. Run this graphql:

```
query NewQuery {
  menuFooter: menuItems(where: {location: FOOTER}) {
    nodes {
      id
      label
    }
  }
  menuHeader:menuItems(where: {location: HEADER}) {
    nodes {
      id
      label
    }
  }
  menuFooterSocial:menuItems(where: {location: FOOTER_SOCIAL}) {
    nodes {
      id
      label
    }
  }
  menuFooterSecondary:menuItems(where: {location: FOOTER_SECONDARY}) {
    nodes {
      id
      label
    }
  }
}
```

Expected behavior would be that only the menu location with a menu assigned has results, however currently all are outputting results.

### WPGraphQL Version

1.13.8

### WordPress Version

6.1.1

### PHP Version

8.0

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by jasonbahl** [comment]
I've tried to reproduce this, and I'm unable. 

I have 2 menu locations, "Primary" and "Footer"

I assigned a menu to "Primary" but left "Footer" unassigned. 


![CleanShot 2023-09-13 at 13 29 55](https://github.com/wp-graphql/wp-graphql/assets/1260765/2f6fe425-2d83-4170-b8fe-88ed3f59d9bb)

I then executed the following query: 

```graphql
{
  primary: menuItems(where: {location: PRIMARY}) {
    nodes {
      id
      menu {
        node {
          id
          name
        }
      }
    }
  }
  footer: menuItems(where: {location: FOOTER}) {
    nodes {
      id
      menu {
        node {
          id
          name
        }
      }
    }
  }
}
```

And I got menuItems returned from the primary menu, but not for the footer menu. 

![CleanShot 2023-09-13 at 13 29 36](https://github.com/wp-graphql/wp-graphql/assets/1260765/73a74612-e7d5-4725-a8d3-3b944e1b520e)

I've tried this on WPGraphQL v1.16.0, v1.14.6 and v1.13.8 and have not been able to reproduce in any version. 

Are you able to reproduce this consistently in _any_ environment? Or is it perhaps an issue with a specific theme or plugin active?
